jabgui/src/main/java/org/jabref/gui/newentry/NewEntryView.java

```diff
@@ -273,6 +270,45 @@ private void initializeLookupIdentifier() {
             idLookupSpecify.selectedProperty().set(true);
         }
 
+        if (!StringUtil.isBlank(clipboardText) && !clipboardText.contains("\n")) {
+            // :TODO: Better validation would be nice here, so clipboard text is only copied over if it matches a
+            // supported identifier format.
+            idText.setText(clipboardText);
+            idText.selectAll();
+
+            Optional<Identifier> identifier = CompositeIdFetcher.getIdentifier(clipboardText);
+            if (identifier.isPresent()) {
+                Identifier id = identifier.get();
+                boolean isValid = switch (id) {
+                    case DOI doi -> DOI.isValid(doi.asString());
+                    case ISBN isbn -> isbn.isValid();
+                    default -> true;
+                };
+
+                if (isValid) {
+                    Platform.runLater(() -> {
+                        idLookupSpecify.setSelected(true);
+                        for (IdBasedFetcher fetcher : idFetcher.getItems()) {
```

Comment:
Move that to a method `fetcherForIdentifier` -- see at the end of the class `fetcherFromName` as another example